rewrite/pyproject.toml

```diff
@@ -1,27 +1,38 @@
-[tool.poetry]
+[build-system]
+requires = ["setuptools>=42", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[project]
 name = "openrewrite"
-version = "0.0.0"  # This will be replaced by the GitHub Actions workflow
 description = "OpenRewrite is a mass automated refactoring engine."
-authors = ["Moderne Inc. <support@moderne.io>"]
-license = "Apache-2.0"
-packages = [
-    { include = "rewrite" }
+version = "0.0.0" # Updated dynamically during release
+authors = [{ name = "Moderne Inc.", email = "support@moderne.io" }]
+license = { text = "Moderne, Inc. Commercial License" }
+dependencies = [
+    "pytest>=8.3.4",
+    "more-itertools>=10.4.0",
+    "openrewrite-remote",
+    "pip>=24.3.1",
+    "pypi-simple>=1.6.1",
 ]
+requires-python = ">=3.12, <4"
 
-[tool.poetry.dependencies]
-python = ">=3.9"
-pytest = "^8.3.2"
```

Comment:
I thought we were standardizing around 3.9 for some reason that now escapes me

Reply:
Yes. I accidentally used 3.12 API to parse f-strings. As soon as we have a customer requiring an older Python interpreter, I will try to fix that.